### PR TITLE
[Practices v2] Add officer/client selection flow

### PR DESCRIPTION
**Pull request by ohaibbq**
## Description of the change
This also upgrades us to `@recidiviz/design-system@3.0.0`.

Screen Shot 2022-03-21 at 4 42 34 PM

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #11352

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -77,60 +79,69 @@ const SHARED_VIEWS = ["", "profile"];
...
-    <SentryErrorBoundary>
-      <Router>
-        <QueryParamProvider ReactRouterRoute={Route}>
-          <Switch>
-            <Route path="/verify" component={VerificationNeeded} />
-            <AuthWall>
+  <ThemeProvider theme={{
+    fonts: {
+      heading: AVAILABLE_FONTS.LIBRE_BASKERVILLE,
+      body: AVAILABLE_FONTS.PUBLIC_SANS,
```
Comment (macfarlandian) [inline src/App.js:85]: do we actually want to apply this everywhere? are older parts of the application still using Libre Franklin?

File: src/PracticesStore/PracticesStore.ts [hunk 2]
```diff
@@ -121,11 +121,25 @@ export class PracticesStore implements Hydratable {
...
+    this.selectedOfficerIds = officerIds;
+  }
+
+  updateSelectedClient(clientId?: string): void {
+    this.selectedClientId = clientId;
+  }
+
+  get selectedOfficers(): StaffRecord[] {
+    return this.availableOfficers.filter(
+      (officer) => this.selectedOfficerIds.indexOf(officer.id) !== -1
```
Comment (macfarlandian) [inline src/PracticesStore/PracticesStore.ts:134]: could also just use `.includes(officer.id)`

**Review by Catacola — COMMENTED** [review]
This all looks good to me, but I'm still running into API key errors when I try to run it locally. I don't have time to try to diagnose why it's broken for just me, so I'll leave it to @macfarlandian for final approval.

File: src/App.js (hunk 1 above)
Comment (ohaibbq) [reply src/App.js:85]: good point! I'll switch this back to Franklin and use another ThemeProvider inside of PracticesV2

**Pull request merged**